### DISCUSSION TITLE
How can i include .kts files to be analyzed

### DISCUSSION THREAD

**Discussion by IvanSanchez-uam** [detekt/detekt#5671] [Q&A] [answered]
I need to create a rule under the gradle file which will analyze if retrofit dependency is included 

dependencies {
    implementation("com.squareup.retrofit2:retrofit:2.9.0")
    implementation("com.squareup.retrofit2:converter-gson:2.9.0")
}

up to I know that is possible to analyze kotlin scrips files (.kts) but the question here is how can I add this files to be analyzed, I try setting this under source configuration but it is not working (**/*.kts)

source = files(
            "src/main/java",
            "src/test/java",
            "src/main/kotlin",
            "src/test/kotlin",
            "**/*.kts"
)

**Comment by schalkms** [comment, marked as answer]
Please try to use `include("**/*.kts")` in your Gradle script.

**Reply by IvanSanchez-uam** [reply]
This is all my gradle file, i set the task as documentation say but im getting this issue:

**A problem occurred evaluating script.
> Could not get unknown property 'io'**

seems like it is not recognizing io.gitlab.arturbosch.detekt.Detek and  o.gitlab.arturbosch.detekt.DetektCreateBaselineTask under the 2 tasks i have.

apply plugin: "io.gitlab.arturbosch.detekt"

detekt {
    toolVersion = Versions.detekt
    config = files("$rootDir/staticAnalysis/detekt-config.yml")
    buildUponDefaultConfig = true

    dependencies {
        detekt("io.gitlab.arturbosch.detekt:detekt-cli:${Versions.detekt}")
        detektPlugins("io.gitlab.arturbosch.detekt:detekt-formatting:${Versions.detekt}")
        detekt(project(":detekt-custom-rules"))
    }
}

tasks.withType(io.gitlab.arturbosch.detekt.Detekt).configureEach {
    include("**/*.kts") 
}

tasks.withType(io.gitlab.arturbosch.detekt.DetektCreateBaselineTask).configureEach {
    include("**/*.kts")
}

**Im usisng 1.20.0 detekt version**

**Reply by schalkms** [reply]
I'm afraid, I'm not sure what you are trying to achieve. The problem statement mentioned in the second comment is different compared to the initial post. Furthermore, I can't reproduce it without more information, unfortunately. 

In my humble opinion, it's best to stick with the examples shown on the detekt website and then customize further based on these examples.
